lib/autoupdate.ps1

```diff
@@ -281,43 +281,90 @@ function get_hash_for_app([String] $app, $config, [String] $version, [String] $u
     return $hash
 }
 
-function update_manifest_with_new_version($json, [String] $version, [String] $url, [String] $hash, $architecture = $null) {
-    $json.version = $version
-
-    if ($null -eq $architecture) {
-        if ($json.url -is [System.Array]) {
-            $json.url[0] = $url
-            $json.hash[0] = $hash
-        } else {
-            $json.url = $url
-            $json.hash = $hash
-        }
-    } else {
-        # If there are multiple urls we replace the first one
-        if ($json.architecture.$architecture.url -is [System.Array]) {
-            $json.architecture.$architecture.url[0] = $url
-            $json.architecture.$architecture.hash[0] = $hash
-        } else {
-            $json.architecture.$architecture.url = $url
-            $json.architecture.$architecture.hash = $hash
+function Update-ManifestProperty {
+    <#
+    .SYNOPSIS
+        Update propert(y|ies) in manifest
+    .DESCRIPTION
+        Update selected propert(y|ies) to given version in manifest.
+    .PARAMETER Manifest
+        Manifest to be updated
+    .PARAMETER Property
+        Selected propert(y|ies) to be updated
+    .PARAMETER AppName
+        Software name
+    .PARAMETER Version
+        Given software version
+    .PARAMETER Substitutions
+        Hashtable of internal substitutable variables
+    .OUTPUTS
+        System.Boolean
+            Flag that indicate if there are any changed properties
+    #>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType([Boolean])]
+    param (
+        [Parameter(Mandatory = $true, Position = 1)]
+        [PSObject]
```

Comment:
To be extact, this should be [PSCustomObject]. PSCustomObject < PSObject < Object, and the first two are both okay. I'll change it to [PSCustomObject]， since no type is not precise enough, I thought.